src/megaapi_impl.cpp

```diff
@@ -25228,21 +25263,42 @@ void MegaFolderUploadController::startFileUploads(const vector<NewNode> &nn)
            vector<LocalPath> &pendingFiles = auxit->second;
            for (size_t j = 0; j < pendingFiles.size(); j++)
            {
-               pendingTransfers++;
                const LocalPath &localpath = pendingFiles.at(j);
                FileSystemType fsType = client->fsaccess->getlocalfstype(localpath);
-               megaApi->startUpload(false, localpath.toPath(*client->fsaccess).c_str(), parentNode.get(), (const char *)NULL, -1, tag, false, NULL, false, false, fsType, this);
-           }
+               MegaTransferPrivate *transfer = megaApi->createUploadTransfer(false, localpath.toPath(*client->fsaccess).c_str(),
```

Comment:
Both this method `createUploadTransfer()` and the line above (`getlocalfstype()`) access to the `MegaClient::fsaccess` from the worker thread. They represent the mutual-exclusion, which needs to be ensured. In other words, we should avoid accessing MegaClient members from both threads, worker and main.